heimdall-core/src/main/java/br/com/conductor/heimdall/core/util/LocalDateTimeDeserializer.java

```diff
@@ -50,44 +53,42 @@ private LocalDateTimeDeserializer(){}
      @Override
      public LocalDateTime deserialize(JsonParser parser, DeserializationContext context) throws IOException {
 
-          switch (parser.getCurrentToken()) {
-               case START_ARRAY:
-                    if (parser.nextToken() == JsonToken.END_ARRAY) {
-                         return null;
-                    }
-                    int year = parser.getIntValue();
-
-                    parser.nextToken();
-                    int month = parser.getIntValue();
-
-                    parser.nextToken();
-                    int day = parser.getIntValue();
-                    
-                    parser.nextToken();
-                    int hour = parser.getIntValue();
-
-                    parser.nextToken();
-                    int minute = parser.getIntValue();
-
-                    parser.nextToken();
-                    int second = parser.getIntValue();
-                    
-                    parser.nextToken();
-                    int nanosecond = parser.getIntValue();
-                    
-                    if (parser.nextToken() != JsonToken.END_ARRAY) {
-                         context.reportWrongTokenException(JsonToken.class, JsonToken.END_ARRAY, "Expected array to end.");
-                    }
-                    return LocalDateTime.of(year, month, day, hour, minute, second, nanosecond);
-
-               case VALUE_STRING:
-                    String string = parser.getText().trim();
-                    if (string.length() == 0) {
-                         return null;
-                    }
-                    return LocalDateTime.parse(string, ISO_DATE_TIME);
+          if (parser.getCurrentToken().equals(START_ARRAY)) {
```

Comment:
I think switch is better in this case, because we are testing against mutiple values of  getCurrentToken() method.

Reply:
ok